### DISCUSSION TITLE
p-tree, infos about the filter-element requested

### DISCUSSION THREAD

**Discussion by MaxZuse** [primefaces/community#1801] [PrimeNG] [answered]
Hello

I use a p-table with the top filter line in Angular. I need to know some simple information about this filtering.

```
<p-tree #tree [value]="nodes" selectionMode="single" [filter]="true" [contextMenu]="cm" [(selection)]="nodesel" scrollHeight="500px"
		(onNodeSelect)="select ($event)" (onNodeContextMenuSelect)="select ($event)" 
		(onNodeExpand)="expand ($event)" (onNodeCollapse)="collapse ($event)">
	</p-tree>
	<p-contextMenu #cm [model]="menus"></p-contextMenu>
```

I have a ViewChild connected to it.

    @ViewChild ("tree") tree;

The only function I call from the component is

    this.tree.resetFilter ()

It works very well - but now I need to add some features.

- Ho do I know if there is input in the filter-textfield - or lets say if filtering is active (I am enabling it with [filter])

- If my [value] changes, the filters output is invalid. It refreshes if I re-filter by changing the input of the filter-element. How can I do a re-filtering manually?

Thanks for support!
Cheers

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Custom filtering is possible with tree, we'll add it as a demo in primeng.org in upcoming days

**Comment by MaxZuse** [comment]
Hello

May I ask whats the status of this. I do not find any new infos at primeng.org.
Thanks!
